WebDriverAgentLib/Categories/XCUIElement+FBTyping.m

```diff
@@ -141,24 +142,33 @@ - (BOOL)fb_clearTextWithSnapshot:(FBXCElementSnapshotWrapper *)snapshot
   NSString *placeholderValue = snapshot.placeholderValue;
   NSUInteger preClearTextLength = [currentValue fb_visualLength];
   do {
-    if (retry >= MAX_CLEAR_RETRIES - 1) {
-      // Last chance retry. Tripple-tap the field to select its content
-
-      if ([self respondsToSelector:@selector(tapWithNumberOfTaps:numberOfTouches:)]) {
-        // e.g. tvOS 17 raised unrecognized selector error for XCUIElementTypeSearchField
-        // while following typeText worked.
-        [self tapWithNumberOfTaps:3 numberOfTouches:1];
-      }
-      return [FBKeyboard typeText:backspaceDeleteSequence error:error];
-    }
-
-    NSString *textToType = [backspaceDeleteSequence fb_repeatTimes:preClearTextLength];
     if (shouldPrepareForInput && 0 == retry) {
       [self fb_prepareForTextInputWithSnapshot:snapshot];
     }
-    if (![FBKeyboard typeText:textToType error:error]) {
+    NSString *backspacesToType = [backspaceDeleteSequence fb_repeatTimes:preClearTextLength];
+#if TARGET_OS_IOS
+    if (retry == MAX_CLEAR_RETRIES - 2) {
+      // https://github.com/appium/appium/issues/19389
+      [[XCUIDevice sharedDevice] fb_performIOHIDEventWithPage:0x07
+                                                        usage:0x9c
+                                                     duration:0.05
+                                                        error:nil];
+    } else if (retry >= MAX_CLEAR_RETRIES - 1) {
+      // Last chance retry. Tripple-tap the field to select its content
+      [self tapWithNumberOfTaps:3 numberOfTouches:1];
+      return [FBKeyboard typeText:backspaceDeleteSequence error:error];
+    } else if (![FBKeyboard typeText:backspacesToType error:error]) {
+      return NO;
+    }
+#else
+    if (retry >= MAX_CLEAR_RETRIES - 1) {
+      return [[[FBErrorBuilder builder]
+               withDescriptionFormat:@"'%@' cannot be cleared of its text", snapshot.fb_description]
+              buildError:error];
+    } else if (![FBKeyboard typeText:backspacesToType error:error]) {
```

Comment:
Can we change here to:
```
    if (retry >= MAX_CLEAR_RETRIES - 1) {
      return [FBKeyboard typeText:backspaceDeleteSequence error:error];
    } else if (![FBKeyboard typeText:backspacesToType error:error]) {
      return NO;
    }
```

?
In the current implementation, calling clear text for an empty search field generates the below error. 

```
> e.clear
Selenium::WebDriver::Error::InvalidElementStateError: Error Domain=com.facebook.WebDriverAgent Code=1 "'XCUIElementTypeSearchField' cannot be cleared of its text" UserInfo={NSLocalizedDescription='XCUIElementTypeSearchField' cannot be cleared of its text}
```

This was because `self.fb_takeSnapshot.value` kept returning its place holder while the field was empty (self.fb_takeSnapshot.label was nil). So we potentially can add `label` comparison as well in addition to `currentValue = self.fb_takeSnapshot.value`... but not sure it is a better idea.

So, just keeping existing behavior with `return [FBKeyboard typeText:backspaceDeleteSequence error:error];` would be safe to keep current clear text's behavior for already cleared field.